Source/JavaScriptCore/runtime/JSGlobalObject.cpp

```diff
@@ -3329,13 +3330,58 @@ void JSGlobalObject::bumpGlobalLexicalBindingEpoch(VM& vm)
 void JSGlobalObject::queueMicrotask(JSFunction* job, JSValue argument0, JSValue argument1, JSValue argument2, JSValue argument3)
 {
     QueuedTask task { nullptr, this, job, argument0, argument1, argument2, argument3 };
+
+    if (m_contextMicrotaskQueue) {
+        m_contextMicrotaskQueue->enqueue(WTFMove(task));
+        return;
+    }
+
     if (globalObjectMethodTable()->queueMicrotaskToEventLoop) {
         globalObjectMethodTable()->queueMicrotaskToEventLoop(*this, WTFMove(task));
         return;
     }
     vm().queueMicrotask(WTFMove(task));
 }
 
+void JSGlobalObject::createContextMicrotaskQueue()
+{
+    ASSERT(!m_contextMicrotaskQueue);
+    m_contextMicrotaskQueue = makeUnique<MicrotaskQueue>(vm());
+}
+
+void JSGlobalObject::drainMicrotasks()
+{
+    if (!m_contextMicrotaskQueue)
+        return;
+        
+    auto& vm = this->vm();
+
+    // TODO: should we be checking drainMicrotaskDelayScopeCount here?
+    // intuitively i think no because the per-context queues are meant
+    // to be drained independently when requested. but i could be wrong.
```

Comment:
After looking into this more I believe my assumption was correct so removing this but leaving the comment so it can be seen just in case.